rllib/models/jax/fcnet.py

```diff
@@ -0,0 +1,125 @@
+        self.free_log_std = model_config.get("free_log_std")
+
+        # Generate free-floating bias variables for the second half of
+        # the outputs.
+        if self.free_log_std:
+            assert num_outputs % 2 == 0, (
+                "num_outputs must be divisible by two", num_outputs)
+            num_outputs = num_outputs // 2
+
+        self._hidden_layers = []
+        prev_layer_size = int(np.product(obs_space.shape))
+        self._logits = None
+
+        # Create layers 0 to second-last.
+        for size in hiddens[:-1]:
+            self._hidden_layers.append(
+                SlimFC(
+                    in_size=prev_layer_size,
+                    out_size=size,
+                    activation_fn=activation))
+            prev_layer_size = size
+
+        # The last layer is adjusted to be of size num_outputs, but it's a
+        # layer with activation.
+        if no_final_linear and num_outputs:
+            self._hidden_layers.append(
+                SlimFC(
+                    in_size=prev_layer_size,
+                    out_size=num_outputs,
+                    activation_fn=activation))
+            prev_layer_size = num_outputs
+        # Finish the layers with the provided sizes (`hiddens`), plus -
+        # iff num_outputs > 0 - a last linear layer of size num_outputs.
+        else:
+            if len(hiddens) > 0:
+                self._hidden_layers.append(
+                    SlimFC(
+                        in_size=prev_layer_size,
+                        out_size=hiddens[-1],
+                        activation_fn=activation))
+                prev_layer_size = hiddens[-1]
+            if num_outputs:
+                self._logits = SlimFC(
+                    in_size=prev_layer_size,
+                    out_size=num_outputs,
+                    activation_fn=None)
+            else:
+                self.num_outputs = (
+                    [int(np.product(obs_space.shape))] + hiddens[-1:])[-1]
+
+        # Layer to add the log std vars to the state-dependent means.
+        if self.free_log_std and self._logits:
+            raise ValueError("`free_log_std` not supported for JAX yet!")
+
+        self._value_branch_separate = None
+        if not self.vf_share_layers:
+            # Build a parallel set of hidden layers for the value net.
+            prev_vf_layer_size = int(np.product(obs_space.shape))
+            vf_layers = []
+            for size in hiddens:
+                vf_layers.append(
+                    SlimFC(
+                        in_size=prev_vf_layer_size,
+                        out_size=size,
+                        activation_fn=activation,
+                    ))
+                prev_vf_layer_size = size
+            self._value_branch_separate = vf_layers
+
+        self._value_branch = SlimFC(
+            in_size=prev_layer_size, out_size=1, activation_fn=None)
+        # Holds the current "base" output (before logits layer).
```

Comment:
Looks like the same thing as Pytroch fcnet. What is the big difference?

Reply:
Using JAX (SlimFC, even though named the same, is a JAX-based module here), not PyTorch.

Reply:
Sounds good.